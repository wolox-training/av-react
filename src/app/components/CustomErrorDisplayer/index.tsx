import React from 'react';

import styles from './styles.module.scss';

interface Props {
  errors: string[];
}

function CustomErrorDisplayer({ errors }: Props) {
  return (
    <div className="column">
      {errors.map(error => (
        <p role="alert" key={error} className={styles.error}>
          {error}
        </p>
      ))}
    </div>
  );
}

export default CustomErrorDisplayer;
