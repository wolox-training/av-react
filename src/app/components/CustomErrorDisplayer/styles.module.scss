@import 'variables/colors';

.error {
  color: $red;
  margin-top: 8px;
}
