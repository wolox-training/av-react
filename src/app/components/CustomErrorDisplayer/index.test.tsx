import React from 'react';
import { render, screen } from '@testing-library/react';

import CustomErrorDisplayer from './';

const errorMsgs = ['Email already taken', 'Password do not match'];
const MAX_ALERTS_LENGTH = 2;

describe('CustomErrorDisplayer test', () => {
  test('Should match snapshot', () => {
    const { container } = render(<CustomErrorDisplayer errors={errorMsgs} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render error messages', () => {
    render(<CustomErrorDisplayer errors={errorMsgs} />);
    expect(screen.getAllByRole('alert').length).toBe(MAX_ALERTS_LENGTH);
  });
});
