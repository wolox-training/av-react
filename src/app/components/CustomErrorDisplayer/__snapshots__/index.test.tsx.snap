// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomErrorDisplayer test Should match snapshot 1`] = `
<div>
  <div
    class="column"
  >
    <p
      class="error"
      role="alert"
    >
      Email already taken
    </p>
    <p
      class="error"
      role="alert"
    >
      Password do not match
    </p>
  </div>
</div>
`;
